### PR TITLE
Log proxy request error

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Logging some metrics from our list

## Proposal

Implementing proxy request error Counter metric. With this and the success one, we'll be able to calculate the error and success rate of the server

## Test Plan

Will add some test code to force some errors, run validator locally, run a benchmark against it, and see if the metric is properly logged

### CODE REVIEW

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1166** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1166" target="_blank">Graphite</a> 
    * **PR #1169** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1169" target="_blank">Graphite</a> 
      * **PR #1170** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1170" target="_blank">Graphite</a> 
        * **PR #1171** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171" target="_blank">Graphite</a> 
          * **PR #1172** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172?utm_source=stack-comment).

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1166** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1166" target="_blank">Graphite</a> 
    * **PR #1169** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1169" target="_blank">Graphite</a> 
      * **PR #1170** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1170" target="_blank">Graphite</a> 
        * **PR #1171** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1171" target="_blank">Graphite</a> 
          * **PR #1172** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172?utm_source=stack-comment).

**Comment by Twey** [comment]
This looks good to me but I'm not sure I understand the Graphite output — do 1166…1171 have to be merged before this can be merged, or does this want to be merged before 1171 can be merged?

**Comment by andresilva91** [comment]
So Graphite is used for stacked PRs. The output is showing the current state of the stack, and the little hand 👈 shows where in the stack we're at.
So #1166 is stacked on top of main (is at the bottom of the stack), #1169 on top of it, and so on. So to merge a PR, you need to also merge all the PRs underneath it in the stack.
I usually land the whole stack at once (Graphite gives that support).

**Comment by Twey** [comment]
Understood, thanks for the clarification!

File: linera-service/src/grpc_proxy.rs [hunk 1]
```diff
@@ -264,7 +279,12 @@ impl ValidatorNode for GrpcProxy {
                     .inc();
                 Ok(chain_info_result)
             }
-            Err(status) => Err(status),
+            Err(status) => {
+                PROXY_REQUEST_ERROR
+                    .with_label_values(&["handle_lite_certificate"])
+                    .inc();
+                Err(status)
+            }
```
Comment (afck) [inline linera-service/src/grpc_proxy.rs:287]: <i class='graphite__hidden'>[Re: lines 276 to 287]<br/></i>

Can we deduplicate this, so we just call:

```rust
Self::handle_proxy_request_outcome(result, "handle_lite_certificate")
```<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by afck — APPROVED** [review]

**Comment by andresilva91** [comment]
### Merge activity

* **Oct 30, 3:05 PM**: @andresilva91 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1172).

**Pull request merged**